### ISSUE TITLE
Dou you plan to instrument bytecode produced by Scala compiler?

### ISSUE THREAD

**Issue by volth** [puniverse/quasar#45]
Hi.
I tried to translate simple Quasar examples from Java to Scala. 
The most simple ones work, but as the complexity increases the code emits "WARNING: Uninstrumented methods on the call stack" and then crashes with NullPointerException or something like this.

It would be nice to have Quasar for Scala (Akka is not the replacement especially for implementing parsers of complex network protocols).
Do you plan to support Scala ?

**Comment by mepcotterell** [comment]
Has there been any more traction with this issue?

**Comment by pron** [comment]
I am afraid not... Other than Java, our focus is currently on Clojure and Kotlin.
I hope that in the near future Quasar will be completely transparent to user code (no need for suspendable annotations), so that it would automatically work with JVM code written in any language. We're working on that, but we're not there yet. Supporting Scala specifically is quite hard as the language is so big, the number of primitive constructs is high, and the compiler is rather impenetrable. Also, Scala developers seem to like the monad approach to asynchronous code.

To sum up: we have not begun work on providing explicit support for Scala, even though this is something we'd like to do in the future. We have begun work on making Quasar completely transparent (so that it would automatically support Scala), but this depends on changes to the JDK so might take some time.

However, if you or anyone else would like to help us with this task, we'll be more than happy to accept it, and promise to provide the necessary guidance.

**Comment by cocodrino** [comment]
the channel system looks very cool, I'd like use them with scala, I really hope than quasar will be a real alternative to typesafe stack, although is hard considering all the advertising made by them...I'm going to suscribe to this issue for future notifications
